File: golang/1d_dp/climbing_stairs.go
package main

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}

	first, second := 1, 2

	for i := 3; i <= n; i++ {
		current := first + second

		first = second
		second = current
	}

	return second
}

File: java/src/main/java/com/sanedge/interview/dp/ClimbingStairs.java
package com.sanedge.interview.dp;

class ClimbingStairs {

    public int climbStairs(int n) {
        if (n <= 2) {
            return n;
        }

        int first = 1;
        int second = 2;

        for (int i = 3; i <= 3; i++) {
            int current = first + second;

            first = second;
            second = current;
        }
        return second;
    }
}
